onnxruntime/core/optimizer/qdq_transformer/selectors_actions/qdq_selectors.cc

```diff
@@ -51,6 +51,14 @@ bool NodeGroupSelector::CheckQDQNodes(const GraphViewer& graph_viewer, const Nod
     return false;
   }
 
+  auto does_node_produce_graph_output = [&graph_viewer](const Node* node_ptr) {
+    return node_ptr == nullptr || graph_viewer.NodeProducesGraphOutput(*node_ptr);
```

Comment:
A comment explaining why nullptr equates to producing graph output would be helpful for future readers of the code. I'm assuming it's because the DQ node is not in the graph_viewer and FindQDQNodes removed it so we don't know for sure.

Reply:
It is to avoid the possible dereference of null_ptr.

Reply:
We could choose to return `false` if it's nullptr though, so I was looking for the reason to choose `true`. 

It looked like FindQDQNodes was the likely cause of the nullptr, but I wouldn't have expected there to be a valid QDQ node group if part of it wasn't in the graph_viewer.

Reply:
If it's nullptr, we need to fail the check finally. That's reason I make it true.